### PR TITLE
Move `ReportError` to `@woocommerce/components` as `AnalyticsError` 

### PR DESCRIPTION
**Pull request by tomalec**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Move ReportError to `@woocommerce/components` as `<AnalyticsError>`
This is the first PR that takes the easiest, lowest-hanging fruit to pave the path for other `<Report*>` components. This one has no dependencies, so should be easier to review so we can focus on the folder structure, etc.

Address part of https://github.com/woocommerce/woocommerce/issues/36052

The goal is to publish those components to be reused by other extensions migrating their legacy reports to the new Analytics.

I decided to rename components to `<Analytics*>` as `@woocommerce/components` already have a `<ReportFilters>` which are actually used inside `plugins/woocommerce-admin/client/analytics/components/report-filters/index.js`.
So the current `@woocommerce/components ReportFilters` are a bit more general-purpose, while `woocommerce-admin/.../analytics... ReportFilters` are specifically for analytics.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout and build this branch
2. Visit any analytics page with an error. You can simulate that by providing malformed query param, like `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fcategories&chart=net_revenue&orderby=net_revenue&order=foo`
3. Build and [check storybook](http://localhost:6007/?path=/story/woocommerce-admin-components-analytics-analyticserror--basic)
```
cd tools/storybook
npm run storybook
```

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Publish `<ReportError>` to `Woocommerce/components`

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: packages/js/components/changelog/dev-move-report-components [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
```
Comment (eason9487) [inline packages/js/components/changelog/dev-move-report-components]: From the perspective of `@woocommerce/components`, would it be a bit closer to the minor level?

**Review by eason9487 — COMMENTED** [review]
Thanks for starting to facilitate the reuse of Analytics component for other extensions! The code works well.

> This one has no dependencies, so should be easier to review so we can focus on the folder structure, etc.

Could you elaborate on the folder structure? I find it a bit different from most other components, e.g. the place of README file and less flat folder level.

I'm assuming that in subsequent PRs, the components under `plugins/woocommerce-admin/client/analytics/components` will be moved to `packages/js/components/src/analytics` and will have their own folder, so for now it seems reasonable to have an extra level, which is `analytics/report-error/index.js` rather than `analytics/report-error.js`. But based on this assumption, I'm also curious if README could be centralized in one file like [Table's](https://github.com/woocommerce/woocommerce/blob/57c13a4c9535be649b8591ce3d6205a93f29965f/packages/js/components/src/table/README.md)?

**Comment by tomalec** [comment]
Thanks for the review @eason9487 !

> Could you elaborate on the folder structure? I find it a bit different from most other components, e.g. the place of README file and less flat folder level. [...]

Sorry, I didn't mention it in the OP. Yes, the idea is to add other components under `/packages/js/components/src/analytics`.

I wanted to group them together under `analytics` to emphasize their desired purpose. Also, they are a bit higher level. So not so, general purpose, dumb, leaf components that you configure and feed with content. They assume a specific data structure of returned by the given endpoint name, etc.

With the Table example, we have one main component `<TableCard>`, plus a few simpler ones to support the table specifically.
In the case of Reports, we do not have one `<Report>` component, or something we could consider a default export of `../analytics/`. `<ReportChart>` or `<ReportTable>` can be use completely independently. That's why I proposed to keep them in separate folders.

Speaking of READMEs, as mentioned above, `<ReportTable>` could be used independently of `<ReportChart>` so I think could benefit from dedicated readmes. Also, I was thinking of making `../analytics/README.md` provide links to individual component READMEs but also links to guidelines for making the serverside counterpart & registering a new Analytics page (pfKYZu-Le-p2).

**Comment by tomalec** [comment]
I converted it to draft, as I would like to try changing it a bit and rename all the components to `<Analytics*>` as `@woocommerce/components` Unfortunately already have a `ReportFilters` component which is used by analytics `ReportFilters` :)
Also, this way, we will distinguish the naming from the legacy reports to analytics.

**Comment by tomalec** [comment]
The `<AnalyticsError>` is at https://github.com/woocommerce/woocommerce/compare/dev/move-report-components-2?expand=1, but maybe I'll publish it for the review as a batch with other components.

**Review by eason9487 — COMMENTED** [review]
Thanks for the explanation about the future adjustments. The change looks good to me.

Not sure if I should mark this PR as approved since it's still in draft status?

**Comment by eason9487** [comment]
By the way, maybe the PR title could be consistent with:

https://github.com/woocommerce/woocommerce/blob/e0340b31f6f3ca6805dd1b3240983e029be139dd/plugins/woocommerce/changelog/36052-dev-move-report-error#L4

**Comment by tomalec** [comment]
> Not sure if I should mark this PR as approved since it's still in draft status?

Sorry, I just changed the status to read for review.

I also added unit tests.

File: packages/js/components/src/analytics/error/tests/index.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		// Use with extreme caution.
+		event.preventDefault();
+	}
+
+	// Mock window.location.reload by using a global variable
+	const originalLocation = window.location;
+
+	beforeAll( () => {
+		// Opt Out of the jsdom error messages
+		window.addEventListener( 'error', onError );
```
Comment (eason9487) [inline packages/js/components/src/analytics/error/tests/index.test.js]: This doesn't seem to be necessary?

File: packages/js/components/src/analytics/error/tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			screen.getByText(
+				'There was an error getting your stats. Please try again.'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'shows reload button', () => {
+		render( <AnalyticsError /> );
+
+		expect( screen.getByText( 'Reload' ) ).toBeInTheDocument();
```
Comment (eason9487) [inline packages/js/components/src/analytics/error/tests/index.test.js]: Would it be a bit better to query the button by role?

```suggestion
		expect(
			screen.getByRole( 'button', { name: 'Reload' } )
		).toBeInTheDocument();
```

File: packages/js/components/src/analytics/error/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	it( 'refreshes the page when Reload Page button is clicked', () => {
+		const reloadMock = jest.fn();
+		Object.defineProperty( window.location, 'reload', {
+			configurable: true,
+			value: reloadMock,
+		} );
+
+		render( <AnalyticsError /> );
+
+		fireEvent.click( screen.getByText( 'Reload' ) );
```
Comment (eason9487) [inline packages/js/components/src/analytics/error/tests/index.test.js]: I would suggest using `userEvent` over `fireEvent` whenever it's able to.

**Review by eason9487 — APPROVED** [review]
LGTM.

Left a few comments about tests but none of them are blockers.

**Pull request merged**